api/v1alpha1/lvmcluster_webhook.go

```diff
@@ -298,23 +300,25 @@ func (v *lvmClusterValidator) verifyDeviceClass(l *LVMCluster) (admission.Warnin
 	return nil, nil
 }
 
-func (v *lvmClusterValidator) verifyPathsAreNotEmpty(l *LVMCluster) error {
+func (v *lvmClusterValidator) verifyPathsAreNotEmpty(l *LVMCluster) (admission.Warnings, error) {
 
 	var deviceClassesWithoutPaths []string
 	for _, deviceClass := range l.Spec.Storage.DeviceClasses {
 		if deviceClass.DeviceSelector != nil {
 			if len(deviceClass.DeviceSelector.Paths) == 0 && len(deviceClass.DeviceSelector.OptionalPaths) == 0 {
-				return ErrPathsOrOptionalPathsMandatoryWithNonNilDeviceSelector
+				return nil, ErrPathsOrOptionalPathsMandatoryWithNonNilDeviceSelector
 			}
 		} else {
 			deviceClassesWithoutPaths = append(deviceClassesWithoutPaths, deviceClass.Name)
 		}
 	}
 	if len(l.Spec.Storage.DeviceClasses) > 1 && len(deviceClassesWithoutPaths) > 0 {
```

Comment:
I would suggest to change this logic so that we use the for loop above to add an admission warning for every deviceClass with a nil device selector. that way we dont need such a complicated if statement here. WDYT?

Reply:
The thing is that this warning is valid only for the single device class scenario. For the multi device class, we already return an error if they do not specify a device selector in L#316.